routers/web/auth/auth.go

```diff
@@ -814,3 +779,25 @@ func ActivateEmail(ctx *context.Context) {
 	// Should users be logged in automatically here? (consider 2FA requirements, etc.)
 	ctx.Redirect(setting.AppSubURL + "/user/settings/account")
 }
+
+func updateSession(ctx *context.Context, deletes []interface{}, updates map[interface{}]interface{}) error {
```

Comment:
```suggestion
type sessionData = map[string]interface{}

func updateSession(ctx *context.Context, keysToDelete []string, updates sessionData) error {
```
We only use strings as key, so for now we could keep this simple. The `sessionData` type is to reduce the noise when calling the method. What do you think?

```go
updateSession(ctx, []string{
	// Delete the openid, 2fa and linkaccount data
	"openid_verified_uri",
	"openid_signin_remember",
	"openid_determined_email",
	"openid_determined_username",
	"twofaUid",
	"twofaRemember",
	"linkAccount",
}, sessionData{
	"uid":   u.ID,
	"uname": u.Name,
})
```

Reply:
It looks better, but I'm not sure if we need to define a new type, maybe `map[string]interface{}` is good enough?

Reply:
Sure, it's just less to type.